.github/workflows/_linux-test.yml

```diff
@@ -272,3 +273,51 @@ jobs:
       - name: Teardown Linux
         uses: pytorch/test-infra/.github/actions/teardown-linux@main
         if: always()
+
+      - name: Check NVIDIA driver installation step
+        if: failure() && steps.install-nvidia-driver.conclusion && steps.install-nvidia-driver.conclusion == 'failure'
```

Comment:
Is this condition too strict? I remember seeing cases where the nvidia drive failed to install during this action, the next time it runs, it could install correctly, at least this has been the case for GPU runners. 
So shutting down this instance forever might be over-acting? Is there a more exact way of identifying a faulty runner?

Reply:
The `setup-nvidia` step has already retried installing NVIDIA driver 3 times https://github.com/pytorch/test-infra/blob/main/.github/actions/setup-nvidia/action.yml#L15-L21, so I'm pretty confident that its failure mean an unrecoverable error.  On the other hand, I think it make sense to gate it further here by running `nvidia-smi` one last time to confirm.  If the command fails, it means the runner is having GPU-related problems for sure